lib/datasource/npm/get.ts

```diff
@@ -93,6 +94,27 @@ export async function getDependency(
     if (res.repository?.directory) {
       dep.sourceDirectory = res.repository.directory;
     }
+
+    // Massage the repository URL for non-compliant strings for github (see issue #4610)
+    // Remove the non-compliant segments of path, so the URL looks like "<scheme>://<domain>/<vendor>/<repo>"
+    // and add directory to the repository
+    const sourceUrlCopy = `${sourceUrl}`;
+    const sourceUrlSplit: string[] = sourceUrlCopy.split('/');
+
+    if (sourceUrlSplit.length > 7 && sourceUrlSplit[2] === 'github.com') {
+      /* istanbul ignore next */
```

Comment:
Please add a test for this instead of ignore

Reply:
Hi @rarkins I've spent some time figuring out how to get the logger output for using in my test, but no luck for now. Could you please give me some clue/advice how you typically test such code blocks without using `istanbul ignore next`?

Reply:
It's not necessary that you check the logger output. Just make sure that the code is passed once you remove the istanbul. i.e. the confirmation of the test will be the code coverage, not any direct test.

Reply:
Thank you, done.